packages/cli/src/zosfiles/create/ds/ds.handler.ts

```diff
@@ -20,11 +20,23 @@ import { generateZosmfOptions } from "../Create.utils";
 
 export default class DataSetHandler extends ZosFilesBaseHandler {
     public async processWithSession(commandParameters: IHandlerParameters, session: AbstractSession): Promise<IZosFilesResponse> {
-        return Create.dataSetLike(
-            session,
-            commandParameters.arguments.dataSetName,
-            commandParameters.arguments.like,
-            generateZosmfOptions(commandParameters.arguments)
-        );
+        if (commandParameters.arguments.dataSetType != undefined) {
+            return Create.dataSet(
+                session,
+                CreateDataSetTypeEnum.DATA_SET_BLANK,
+                commandParameters.arguments.dataSetName,
+                generateZosmfOptions(commandParameters.arguments)
+            );
+
+        }
+        else {
+            return Create.dataSetLike(
+                session,
+                commandParameters.arguments.dataSetName,
+                commandParameters.arguments.like,
+                generateZosmfOptions(commandParameters.arguments)
+            );
+        }
+
```

Comment:
For the logic to choose between `Create.dataSet` or `Create.dataSetLike`, I don't think it should be based on the `--data-set-type` argument. There are a number of other arguments such as `--primary-space` or `--record-length` that could also be passed to the command without the `--like` option.

I would suggest using the following logic:
```javascript
if (commandParameters.arguments.like == null) {
  // Create.dataSet
} else {
  // Create.dataSetLike
}
```

Reply:
Good point, I've changed the logic to use `commandParameters.arguments.like == null`.